src/Model/Customer/Customer.php

```diff
@@ -0,0 +1,303 @@
+        return $new;
+    }
+
+    /**
+     * @return CustomerDeliveryAddress
+     */
+    public function getDeliveryAddress()
+    {
+        return $this->deliveryAddress;
+    }
+
+    /**
+     * @param CustomerDeliveryAddress $deliveryAddress
+     * @return Customer
+     */
+    public function withDeliveryAddress(CustomerDeliveryAddress $deliveryAddress)
+    {
+        $new = clone $this;
+        $new->deliveryAddress = $deliveryAddress;
+        return $new;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCompanyType()
+    {
+        return $this->companyType;
+    }
+
+    /**
+     * @param string $companyType
+     * @return Customer
+     */
+    public function withCompanyType(string $companyType)
+    {
+        $new = clone $this;
+        $new->companyType = $companyType;
+        return $new;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getCreatedAt(): \DateTime
+    {
+        return $this->createdAt;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getUpdatedAt(): \DateTime
+    {
+        return $this->updatedAt;
+    }
+
+    /**
+     * @param \DateTime $updatedAt
+     */
+    public function setUpdatedAt(\DateTime $updatedAt)
+    {
+        $this->updatedAt = $updatedAt;
+    }
+
+    public static function createFromArray(array $data): Customer
+    {
+
+        $customer = new self();
+        $contact = new  CustomerContact($data['data']['contact']['name'], $data['data']['contact']['email'], $data['data']['contact']['phone']);
+        $address= new CustomerBillingAddress($data['data']['address']['careof'], $data['data']['address']['use_careof_as_attention'], $data['data']['address']['street_address'], $data['data']['address']['zipcode'], $data['data']['address']['city'], $data['data']['address']['country']);
+        $deliveryAddress = new CustomerDeliveryAddress($data['data']['delivery_address']['name'], $data['data']['delivery_address']['street_address'], $data['data']['delivery_address']['careof'], $data['data']['delivery_address']['zipcode'], $data['data']['delivery_address']['city'], $data['data']['delivery_address']['country']);
+        $customer->customerNo = $data['data']['customer_no'] ?? null;
+        $customer->name = $data['data']['name'] ?? null;
+        $customer->notes = $data['data']['notes'] ?? null;
+        $customer->orgNo = $data['data']['org_no'] ?? null;
+        $customer->vatNo = $data['data']['vat_no'] ?? null;
+        $customer->contact= $contact;
+        $customer->address = $address;
+        $customer->deliveryAddress = $deliveryAddress;
+        $customer->createdAt = $data['data']['created_at'];
+        $customer->updatedAt = $data['data']['updated_at'];
+        $customer->companyType = $data['data']['company_type'];
+         return $customer;
+    }
+
+    public function toArray()
+    {
+        //$data['customer_no'] = $this->orgNo ?? null;
+        $data['name'] = $this->name ?? null;
```

Comment:
If  $this->name  is null we should not have a `$data['name']`. Same with all others.